### PR TITLE
Rangedattack

### PR DESCRIPTION
**Pull request by lokiy999**
This PR will add a ranged attack to the player and add a HUD element of the held weapon in the upper left corner.

### CODE REVIEW

File: Assets/Scripts/UI/HUDController.cs [hunk 1]
```diff
@@ -8,13 +8,19 @@ namespace OperationBlackwell.UI {
 	public class HUDController : MonoBehaviour {
 
 		[SerializeField] private Text ActionPointsText_;
+		[SerializeField] private Text WeaponText_;
```
Comment (Westlanderz) [inline Assets/Scripts/UI/HUDController.cs:11]: Variables should be `camelCase` I know the previous var is not that either so I ask if you could change both of them.

**Review by Westlanderz — CHANGES_REQUESTED** [review]
LGTM except for some styling changes.

**Review by Dennisbonke — CHANGES_REQUESTED** [review]
I assume `KeyCode.Alpha1` and `KeyCode.Alpha2` 1 and 2 on the number row? Please add that to the help text in the tutorial level. Also, I see a merge commit and that should be removed.

File: Assets/Scripts/UI/HUDController.cs [hunk 2]
```diff
@@ -7,14 +7,20 @@
...
+		[SerializeField] private Text actionPointsText_;
+		[SerializeField] private Text weaponText_;
 
 		private void Awake() {
 			GridCombatSystem.Instance.OnUnitActionPointsChanged += UpdateActionPoints;
+			GridCombatSystem.Instance.OnWeaponChanged += UpdateWeapon;
 		}
 
 		private void UpdateActionPoints(object sender, GridCombatSystem.UnitEvent args) {
 			ActionPointsText_.text = "Action Points: " + args.unit.GetActionPoints().ToString();
```
Comment (Westlanderz) [inline Assets/Scripts/UI/HUDController.cs:19]: Now this is broken code as this variable does not exist here.

File: Assets/Scripts/UI/HUDController.cs [hunk 3]
```diff
@@ -7,14 +7,20 @@
...
 		}
+
+		private void UpdateWeapon(object sender, string name) {
+			WeaponText_.text = "Current Weapon: " + name;
```
Comment (Westlanderz) [inline Assets/Scripts/UI/HUDController.cs:23]: Same here.

**Review by Westlanderz — CHANGES_REQUESTED** [review]

**Review by Westlanderz — CHANGES_REQUESTED** [review]
The files are correct but you still have a merge commit in the commit list, you must delete this, ask the git master.

**Pull request closed without merging**